packages/flutter_tools/lib/src/build_runner/build_script.dart

```diff
@@ -394,7 +403,9 @@ import 'dart:ui' as ui;
 import "${path.url.basename(buildStep.inputId.path)}" as entrypoint;
 
 Future<void> main() async {
-  await ui.webOnlyInitializePlatform();
+  if ($initializePlatform) {
```

Comment:
doesn't really matter as it will be tree shaken anyway but you could write
${initializePlatform ? 'await ui.webOnlyInitializePlatform();' : ''}
to generate code with the initialize line omitted depending on the style you are going for with this generated code.